scripts/labels/label_tool/checker_labels.py

```diff
@@ -0,0 +1,137 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+"""Provides I/O with the configuration files that describe checker labels."""
+from collections import deque
+import json
+import pathlib
+from typing import Dict, List, Optional, cast
+
+from .output import Settings as OutputSettings, error, trace
+
+
+_ConfigFileLabels = Dict[str, List[str]]
+
+SingleLabels = Dict[str, Optional[str]]
+Labels = Dict[str, Dict[str, str]]
+
+
+def _load_json(path: pathlib.Path) -> Dict:
+    try:
+        with path.open("r") as file:
+            return json.load(file)
+    except OSError:
+        import traceback
+        traceback.print_exc()
+
+        error("Failed to open label config file '%s'", path)
+        raise
+    except json.JSONDecodeError:
+        import traceback
+        traceback.print_exc()
+
+        error("Failed to parse label config file '%s'", path)
+        raise
+
+
+def _save_json(path: pathlib.Path, data: Dict):
+    try:
+        with path.open("w") as file:
+            json.dump(data, file, indent=2)
+            file.write('\n')
+    except OSError:
+        import traceback
+        traceback.print_exc()
+
+        error("Failed to write label config file '%s'", path)
+        raise
+    except (TypeError, ValueError):
+        import traceback
+        traceback.print_exc()
+
+        error("Failed to encode label config file '%s'", path)
+        raise
+
+
+class MultipleLabelsError(Exception):
+    """
+    Raised by `get_checker_labels` if multiple labels exist for the same key.
+    """
+
+    def __init__(self, key):
+        super().__init__("Multiple labels with key: %s", key)
+        self.key = key
+
+
+def get_checker_labels(analyser: str, path: pathlib.Path, key: str) \
+        -> SingleLabels:
+    """
+    Loads and filters the checker config label file available at `path`
+    for the `key` label. Raises `MultipleLabelsError` if there is at least
+    two labels with the same `key`.
+    """
+    try:
+        label_cfg = cast(_ConfigFileLabels, _load_json(path)["labels"])
+    except KeyError:
+        error("'%s' is not a label config file", path)
+        raise
+
+    prefix = f"{key}:"
```

Comment:
The format of the label files allows whitespaces around the colon (`:`) character and at the two ends of the label. Currently we don't have label files containing whitespaces in labels, but at least a TODO may be worth here. Another option could be prohibiting whitespaces in label files.

Reply:
So, just to clarify, `" foo : bar"` as a label **is** the most extreme but still valid format? A.k.a., the "label" is the full `key:value` string in your interpretation.

I think it will be easy to have this tool support that.

Reply:
I wasn't sure about its necessity when I added this many `strip()`s, but currently this is the implementation:
https://github.com/Ericsson/codechecker/blob/11554c62c8642af6ad3f35514aff41598d080b25/codechecker_common/checker_labels.py#L82-L95

Reply:
@bruntib Thanks for the code link! If I come up with a simpler solution I'll just include changes to this file as part of this patch as well!

But due to the massively non-ergonomic and unusual complexity of creating a working Python package out of CodeChecker's source, I'd like not to rely on loading stuff directly from the main source tree.

Alternatively, I'll just have to say you can't use these scripts prior to having run `make package` and that's it...